### PR TITLE
Added completion for dart cli

### PR DESCRIPTION
**Pull request by tbhaxor**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: completion/available/dart.completion.bash [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# shellcheck shell=bash
+
+__dart_completion() {
+	local prev=$(_get_pword)
+	local curr=$(_get_cword)
+
+	local HELP="--help -h"
+	local VERBOSE="-v --verbose"
+	local DEFAULT="$BASE --authtoken --region"
```
Comment (NoahGorny) [inline completion/available/dart.completion.bash:9]: you did not define `$BASE`, also `DEFAULT` is unused

File: completion/available/dart.completion.bash [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			;;
+		pub)
+			# shellcheck disable=SC2207
+			COMPREPLY=($(compgen -W "$HELP $VERBOSE --version --no-trace --trace --verbosity cache deps downgrade get global logout outdated publish run upgrade uploader version" -- "$curr"))
+			;;
+		run)
+			# shellcheck disable=SC2207
+			COMPREPLY=($(compgen -W "$HELP --observe --enable-vm-service --no-pause-isolates-on-exit --no-pause-isolates-on-unhandled-exceptions --no-warn-on-pause-with-no-debugger --pause-isolates-on-exit --pause-isolates-on-unhandled-exceptions --warn-on-pause-with-no-debugger" -- "$curr"))
+			;;
+		*)
```
Comment (NoahGorny) [inline completion/available/dart.completion.bash:40]: please do like I did, and complete all the options only for the command itself (`dart`), in the default case, dont complete anything

**Review by NoahGorny — CHANGES_REQUESTED** [review]
hey @tbhaxor, I added some comments, you are welcome to take a look

nice job :smile:

**Comment by tbhaxor** [comment]
@NoahGorny I will fix this in this weekend

**Review by tbhaxor — COMMENTED** [review]
@NoahGorny This is done

**Comment by tbhaxor** [comment]
@NoahGorny Its done plz chk

**Review by NoahGorny — APPROVED** [review]
Lgtm

**Comment by tbhaxor** [comment]
> I want to add nested completion for `pub` command
> 
> ```
> $ dart pub <TAB> <TAB>
> cache deps downgrade get global logout outdated publish run upgrade uploader version
> ```
> 
> How can I achieve this?
> 
> Thanks in advance

@NoahGorny ping

**Comment by NoahGorny** [comment]
> > I want to add nested completion for `pub` command
> > ```
> > $ dart pub <TAB> <TAB>
> > cache deps downgrade get global logout outdated publish run upgrade uploader version
> > ```
> > 
> > 
> > How can I achieve this?
> > Thanks in advance
> 
> @NoahGorny ping

I dont understand what you mean, this is my output:
```
$ dart pub <TAB> <TAB>
cache        downgrade    global       --help       --no-trace   publish      --trace      uploader     --verbose    --version
deps         get          -h           logout       outdated     run          upgrade      -v           --verbosity  version
```

This is not what you want?

**Comment by tbhaxor** [comment]
@NoahGorny My bad, I wasn't clear enough

See, we have completed for dart, and its arguments like `dart pub`. But how to add the completion for `dart pub add`


image

I need to add the completions for `dart pub add` somehow in `pub` case

**Comment by tbhaxor** [comment]
@NoahGorny Also for such KB, it's hard to search through pull requests. Should we start GitHub discussions for such things?

**Comment by tbhaxor** [comment]
@NoahGorny Ping 😄

**Comment by NoahGorny** [comment]
> @NoahGorny My bad, I wasn't clear enough
> 
> See, we have completed for dart, and its arguments like `dart pub`. But how to add the completion for `dart pub add`
> 
> image
> 
> I need to add the completions for `dart pub add` somehow in `pub` case

You can use the `COMP_WORDS` variable for that, see what we do in `bash-it.completion.bash`, there are nice examples there

**Comment by NoahGorny** [comment]
I will merge this as is- after I will release a stable version today

**Pull request merged**